### ISSUE TITLE
material-input does not accept ngModel like numbers (num, int, double) in angular_components 0.9.0-alpha+3

### ISSUE THREAD

**Issue by supermuka** [dart-lang/angular_components#214]
**Environment:**

```
Windows 7

SDK 2.0.0-dev.19.0

dependencies:
  angular: 5.0.0-alpha+4
  angular_components: 0.9.0-alpha+3
  angular_forms: 1.0.1-alpha+4

builders:
  angular:
    options:
      use_new_template_parser: True

web:
   compiler:
     debug: dartdevc
```

In previous version Angular_Components 0.8.0 material-input like numbers  (num, int, double) works.

But in Angular_components 0.9.0-alpha+3 it does not work.

**HTML template**
```
  <material-input label="Number?"
                  type="number"
                  autoFocus floatingLabel style="width:80%"
                  [(ngModel)]="newTodo2">
  </material-input>
```

**Dart code**
```
class TodoListComponent implements OnInit {
...
  num newTodo2;
...
}
```

The following error is occuring:

```
EXCEPTION: Type 'String' is not a subtype of type 'num'
STACKTRACE: 
dart:core                                                                                  check_num
package:untitled5/src/todo_list/todo_list_component.template.dart 372:17                   [_handle_ngModelChange_2_0]
package:angular/src/core/linker/app_view.dart 624:60                                       src__core__linker__app_view_utils.appViewUtils.eventManager.getZone.runGuarded.dart.fn
dart:async                                                                                 run
package:angular/src/core/zone/ng_zone.dart 175:20                                          [_run]
dart:async                                                                                 runGuarded
package:angular/src/core/zone/ng_zone.dart 311:22                                          runGuarded
package:angular/src/core/linker/app_view.dart 624:19                                       event
dart:async                                                                                 runUnary
package:angular/src/core/zone/ng_zone.dart 185:20                                          [_runUnary]
dart:async                                                                                 add
package:angular_forms/src/directives/ng_model.dart 142:12                                  viewToModelUpdate
package:angular_forms/src/directives/shared.dart 30:8                                      dir.valueAccessor.registerOnChange.dart.fn
dart:_runtime                                                                              dcall
package:angular_components/material_input/material_input_default_value_accessor.dart 25:7  value
dart:async                                                                                 runUnary
package:angular/src/core/zone/ng_zone.dart 185:20                                          [_runUnary]
dart:async                                                                                 add
package:angular_components/material_input/base_material_input.dart 393:24                  inputKeypress
package:angular_components/material_input/material_input.template.dart 421:8               [_handle_input_8_3]
package:angular/src/core/linker/app_view.dart 624:60                                       src__core__linker__app_view_utils.appViewUtils.eventManager.getZone.runGuarded.dart.fn
dart:async                                                                                 run
package:angular/src/core/zone/ng_zone.dart 175:20                                          [_run]
dart:async                                                                                 runGuarded
package:angular/src/core/zone/ng_zone.dart 311:22                                          runGuarded
package:angular/src/core/linker/app_view.dart 624:19                                       event


-----async gap-----
dart:async                                                                listen
package:untitled5/src/todo_list/todo_list_component.template.dart 191:40  build
package:angular/src/core/linker/app_view.dart 225:12                      create
package:angular/src/debug/debug_app_view.dart 73:7                        create
package:untitled5/app_component.template.dart 87:16                       build
package:angular/src/core/linker/app_view.dart 225:12                      create
package:angular/src/debug/debug_app_view.dart 73:7                        create
package:untitled5/app_component.template.dart 290:16                      build
package:angular/src/core/linker/app_view.dart 235:12                      createHostView
package:angular/src/debug/debug_app_view.dart 88:7                        createHostView
package:angular/src/core/linker/component_factory.dart 88:20              create
package:angular/src/core/application_ref.dart 415:37                      src__core__linker__component_factory.ComponentRef$._check.run.dart.fn
package:angular/src/core/application_ref.dart 383:18                      dart.dsend.dart.fn
dart:async                                                                run
package:angular/src/core/zone/ng_zone.dart 175:20                         [_run]
dart:async                                                                run
package:angular/src/core/zone/ng_zone.dart 305:22                         run
dart:_runtime                                                             dsend
package:angular/src/core/application_ref.dart 381:1                       run
package:angular/src/core/application_ref.dart 413:12                      bootstrap
package:angular/src/core/application_ref.dart 93:18                       <fn>
dart:async                                                                runUnary
package:angular/src/core/zone/ng_zone.dart 185:20                         [_runUnary]

```

**Comment by supermuka** [comment]
I found how around this problem.

If the following instructions in pubspec.yaml are comments, works fine!

```
#web:
#   compiler:
#     debug: dartdevc
```

**Comment by TedSander** [comment]
@nshahan looks like the [uber directive list](https://github.com/dart-lang/angular_components/blob/d700a88143ed3a177271c3a356f19fbcdef0d7fd/lib/angular_components.dart#L175) doesn't include the number accessor [here](https://github.com/dart-lang/angular_components/blob/d700a88143ed3a177271c3a356f19fbcdef0d7fd/lib/material_input/material_number_accessor.dart#L22)

As a work around you should be able to add materialNumberInputDirectives to your directives list.

**Comment by nshahan** [comment]
Good catch. I'll get that added to the MaterialDirectives list in the next push.

**Comment by nshahan** [comment]
Fixed in 05a16bd7f99954375574304522728bd8cff8bb83.
